### ISSUE TITLE
Restoring a 10h old data, caused channels in indefinite stateis recovery possible? Version 0.5.2

### ISSUE THREAD

**Issue by MykelSIlver** [lightningnetwork/lnd#2757]
### Background

Like issue #2578, but @halseth stated:

> The next version of LND will make sure this cannot be done for channels that have had their state lost.

More a general question than a bug, and this is an exceptional circumstance.
### Your environment

* version of `lnd`
`lnd version 0.5.2-beta commit=v0.5.2-beta`
* which operating system (`uname -a` on *Nix)
`Linux 4.19.5 #24 SMP PREEMPT Mon Feb 18 05:24:26 UTC 2019 aarch64 GNU/Linux`
* version of `bitcoind`
`Bitcoin Core Daemon version v0.17.1`

### Steps to reproduce
Because of a crash a I had to restore a backup from 10 hours earlier. No tx were done during this period. However six channel close channel transactions were made during the crash causing the channels an indefinite state after restoring the old backup. 

### Funds lost:
About 1500 euro (0,44945732 BTC)

The following channels are in an indefinite state.
(1)
Channelpoint: [b06ad896e955ab3fd92a870eafdac61ab300226492edcc800be2c27626778211](b06ad896e955ab3fd92a870eafdac61ab300226492edcc800be2c27626778211)

I can see the [transaction](https://blockstream.info/tx/d5c897ebe4e53343ee12a52188b6abde362e1448a51d198e8dd5907540b40f4d) 
Funds are moved into [this](https://blockstream.info/address/bc1qt3gpq9p2z7w337yhlyge77w2czzfxgyvu79x2l) unspend address

This is my address! But because of the restore of the old backup my node is not able to see this.
How can I fix this?

Other channels in an indefinite state are:
(2)
Channelpoint: [ca59f20c8a371678c4e2abaab4fbc0123e864cb2d39c0391c12afb120174b9b3](https://blockstream.info/tx/ca59f20c8a371678c4e2abaab4fbc0123e864cb2d39c0391c12afb120174b9b3)
(3)
Channelpoint: [4be8b1fd780739f20901d6bb0a6f3e17510026efbcf6b62f051463fa32dcb312](https://blockstream.info/tx/4be8b1fd780739f20901d6bb0a6f3e17510026efbcf6b62f051463fa32dcb312)
(4)
Channelpoint: [f3665b0c3c07cc0946f0eb37cf40e595f3deaac34c4bd7c6f1f4b1929880db08](https://blockstream.info/tx/f3665b0c3c07cc0946f0eb37cf40e595f3deaac34c4bd7c6f1f4b1929880db08)
(5)
Channelpoint: [b8e7a58656a00b85280a97a87e7f5455392dedefc7bb3089813d90b971ff9473](https://blockstream.info/tx/b8e7a58656a00b85280a97a87e7f5455392dedefc7bb3089813d90b971ff9473)
(6)
Channelpoint: [b06ad896e955ab3fd92a870eafdac61ab300226492edcc800be2c27626778211](https://blockstream.info/tx/b06ad896e955ab3fd92a870eafdac61ab300226492edcc800be2c27626778211)

Again I made this new issue because I read that @halseth stated:

> The next version of LND will make sure this cannot be done for channels that have had their state lost.
Well, I am using the 0.5.2 version of the lnd daemon, which is the next version.....

So is there a change that after a while my pending waiting_close_channels channels in the indefinite state will be processed at a certain time?

Or are in my case the funds lost forever?

Thanks in advance

**Comment by Roasbeef** [comment]
You should never restore from an out dated state. If you did and closed those channels out, then you may have lost those funds. 

> No tx were done during this period
Fee updates can still happen automatically. By next version he meant 0.6 not 0.5.2.

**Comment by Roasbeef** [comment]
You can also look at `lncli closedchannels` to see _how_ the channels were closed. If the remote peer detected the "local data loss mode", then if they're `lnd` nodes 0.5.2 and beyond, they'll send you the data to recover those funds. If they aren't (or another implementation all together), then they may not.

**Comment by MykelSIlver** [comment]
> You should never restore from an out dated state. If you did and closed those channels out, then you may have lost those funds.
> 
> > No tx were done during this period
> > Fee updates can still happen automatically. By next version he meant 0.6 not 0.5.2.

The other party did close the channels. Not me. Makes that a difference?

**Comment by MykelSIlver** [comment]
>  If the remote peer detected the "local data loss mode", then if they're `lnd` nodes 0.5.2 and beyond, they'll send you the data to recover those funds. If they aren't (or another implementation all together), then they may not.

So the remote peer also have to have version 0.5.2 and beyond?

**Comment by MykelSIlver** [comment]
Is it safe to run [dropwtxmgr](https://github.com/btcsuite/btcwallet/blob/74a7124666b49cd7e67da87c61169f4fb83fbd73/docs/force_rescans.md) to do a rescan on the 0.5.2. lnd wallet database?
Can this solve my problem?

**Comment by MykelSIlver** [comment]
So it makes no sense to make backups. When something happens, you always restore an old state (backups are at least some hours old). When nodes close channels when your node is offline, and this is the case when you restore the backup, then those funds are lost.

**Comment by halseth** [comment]
Yes, making regular disk backups make no sense for LN nodes.

What is likely the case here is that the channels that are in the "waiting_close" section are to nodes without data loss protection.... and most likely lost :/

For the other channel, how do you know it is your address? Running droptxmanager to rescan (or start a new node from your seed) should restore it then.

**Comment by MykelSIlver** [comment]
Thanks for your reply. Nodes in my waiting_close section are pretty big and known and famous ones: yalls.org, ACINQ, 1ML.com node ALPHA. So maybe they run the most up to date daemon. 
But a restart of my daemon gives me little hope:
```
ChannelPoint(b06ad896e955ab3fd92a870eafdac61ab300226492edcc800be2c27626778211:0) has status Unknown(00000110), won't start.
ChannelPoint(ca59f20c8a371678c4e2abaab4fbc0123e864cb2d39c0391c12afb120174b9b3:0) has status Unknown(00000110), won't start.
ChannelPoint(4be8b1fd780739f20901d6bb0a6f3e17510026efbcf6b62f051463fa32dcb312:0) has status Unknown(00000110), won't start.
Unprompted commitment broadcast for ChannelPoint(f3665b0c3c07cc0946f0eb37cf40e595f3deaac34c4bd7c6f1f4b1929880db08:0) 
ChannelPoint(b8e7a58656a00b85280a97a87e7f5455392dedefc7bb3089813d90b971ff9473:1) has status Unknown(00000110), won't start.
ChannelPoint(b06ad896e955ab3fd92a870eafdac61ab300226492edcc800be2c27626778211:0) has status Unknown(00000110), won't start.

```
For the other channel I was making an assumption. My apologize.  I'm quite upset for what has happened. For me lots of time and funds are gone. But i'm glad you give me some useful info. Maybe I'll try to start a new node from my for a restore! That's a good one!

When I start a new node from from seed, do I need to stop the old one? I rather do not take offline my node too much, but if i start a new node from my seed it maybe conflicts with the running one?

Awaiting your reply and thnak you for your patience!

**Comment by Roasbeef** [comment]
If you're starting a new node from seed, then you should take down the old node.

**Comment by Roasbeef** [comment]
For some of those nodes, it may also be the case that they haven't fully swept their funds yet, so they haven't marked the channel in the final closing state. As for ACINQ, I don't think they fully implement the protocol reliably like we do atm.

**Comment by MykelSIlver** [comment]
Thank you. For now I think I have to have some patience. I'll give it 2000 blocks time before running into conclusions and close this issue after that. Of course I keep this thread updated when things happen.

**Comment by MykelSIlver** [comment]
> If you're starting a new node from seed, then you should take down the old node.

Okay, just one thing before I begin. I take down the old node. But it is not necessary to close all channels for the current node I hope? I imagine the new node is an empty database with just only the funds?

> (A voice in my head tells me this experiment does not have a big change of success because my current node also has the same seed and still receive and send funds of my current channels, but maybe i see it wrong....)

**Comment by Roasbeef** [comment]
This might be related to #2766 (fixes an off by one error, if you update to this, then some of the outputs may be properly swept).

**Comment by Roasbeef** [comment]
Closing for now since we've received no updates, and it appears to have been fixed by the recent PR I linked above (is part of 0.6).

**Comment by MykelSIlver** [comment]
updated to version: lnd version 0.6.0-beta commit=v0.6-beta-rc4
Unfortunately channels still pending at version?
Should a new issue report be made?

after starting the lnd daemon and finishing its startup pending channels gives still:

```
{
    "total_limbo_balance": "44948535",
    "pending_open_channels": [
    ],
    "pending_closing_channels": [
    ],
    "pending_force_closing_channels": [
        {
            "channel": {
                "remote_node_pub": "03c436af41160a355fc1ed230a64f6a64bcbd2ae50f12171d1318f9782602be601",
                "channel_point": "cf91a4e77e81cdec69eb2fa94f8d15a4235cf60de06380a2aec4b743abd99091:1",
                "capacity": "250000",
                "local_balance": "2803",
                "remote_balance": "0"
            },
            "closing_txid": "5a4d2eb780c0c4d3b07d9de57ed47411eed7bd9bfc65abc2fd59e747075a5a89",
            "limbo_balance": "2803",
            "maturity_height": 566904,
            "blocks_til_maturity": -4391,
            "recovered_balance": "0",
            "pending_htlcs": [
            ]
        }
    ],
    "waiting_close_channels": [
        {
            "channel": {
                "remote_node_pub": "0217890e3aad8d35bc054f43acc00084b25229ecff0ab68debd82883ad65ee8266",
                "channel_point": "b8e7a58656a00b85280a97a87e7f5455392dedefc7bb3089813d90b971ff9473:1",
                "capacity": "10000000",
                "local_balance": "9994349",
                "remote_balance": "0"
            },
            "limbo_balance": "9994349"
        },
        {
            "channel": {
                "remote_node_pub": "023f61eb4abd39ee37190fcaf6343050b0b70da647960f9a95e9974cebc4fffd47",
                "channel_point": "b06ad896e955ab3fd92a870eafdac61ab300226492edcc800be2c27626778211:0",
                "capacity": "16777215",
                "local_balance": "16770786",
                "remote_balance": "0"
            },
            "limbo_balance": "16770786"
        },
        {
            "channel": {
                "remote_node_pub": "02a35e94b403dead4d08a00e3df3577682e34f5c4ba39476497be838dc76d15a09",
                "channel_point": "a9c6512546094d3615587d1375459b516a43fa6f40aefb8d5e94211a1b998496:0",
                "capacity": "50000",
                "local_balance": "0",
                "remote_balance": "0"
            },
            "limbo_balance": "0"
        },
        {
            "channel": {
                "remote_node_pub": "0317d6b11e81729192aab651174991a32e566786a7abb45b88898ed27e218a15e6",
                "channel_point": "f3665b0c3c07cc0946f0eb37cf40e595f3deaac34c4bd7c6f1f4b1929880db08:0",
                "capacity": "1500000",
                "local_balance": "1319575",
                "remote_balance": "0"
            },
            "limbo_balance": "1319575"
        },
        {
            "channel": {
                "remote_node_pub": "03864ef025fde8fb587d989186ce6a4a186895ee44a926bfc370e2c366597a3f8f",
                "channel_point": "4be8b1fd780739f20901d6bb0a6f3e17510026efbcf6b62f051463fa32dcb312:0",
                "capacity": "100000",
                "local_balance": "89777",
                "remote_balance": "0"
            },
            "limbo_balance": "89777"
        },
        {
            "channel": {
                "remote_node_pub": "03e50492eab4107a773141bb419e107bda3de3d55652e6e1a41225f06a0bbf2d56",
                "channel_point": "ca59f20c8a371678c4e2abaab4fbc0123e864cb2d39c0391c12afb120174b9b3:0",
                "capacity": "16777215",
                "local_balance": "16771245",
                "remote_balance": "0"
            },
            "limbo_balance": "16771245"
        }
    ]
}

```

Studying the lnd.log filtered for the pending channels gives:
```
2019-04-12 08:26:20.194 [INF] LTND: Version: 0.6.0-beta commit=v0.6-beta-rc4, build=production, logging=default
2019-04-12 08:26:20.194 [INF] LTND: Active chain: Bitcoin (network=mainnet)
2019-04-12 08:26:20.208 [INF] CHDB: Checking for schema update: latest_version=8, db_version=7
2019-04-12 08:26:20.209 [INF] CHDB: Performing database schema migration
2019-04-12 08:26:20.209 [INF] CHDB: Applying migration #8
2019-04-12 08:26:20.209 [INF] CHDB: Migrating to the gossip message store new key format
2019-04-12 08:26:20.209 [INF] CHDB: Migration to the gossip message store new key format complete!

2019-04-12 08:26:53.918 [INF] SWPR: Candidate sweep set of size=1, has yield=0.09988237 BTC
2019-04-12 08:26:53.918 [INF] SWPR: Sweep candidates at height=571294, yield 1 distinct txns
2019-04-12 08:26:53.929 [INF] CNCT: Creating ChannelArbitrators for 1 closing channels
2019-04-12 08:26:53.929 [INF] NTFN: New block epoch subscription
2019-04-12 08:26:53.930 [INF] NTFN: New spend subscription: spend_id=8, outpoint=b06ad896e955ab3fd92a870eafdac61ab300226492edcc800be2c27626778211:0, height_hint=566232
2019-04-12 08:26:53.930 [INF] NTFN: New spend subscription: spend_id=5, outpoint=b8e7a58656a00b85280a97a87e7f5455392dedefc7bb3089813d90b971ff9473:1, height_hint=566232
2019-04-12 08:26:53.930 [INF] NTFN: New spend subscription: spend_id=6, outpoint=4be8b1fd780739f20901d6bb0a6f3e17510026efbcf6b62f051463fa32dcb312:0, height_hint=566233
2019-04-12 08:26:53.930 [INF] NTFN: New spend subscription: spend_id=4, outpoint=f3665b0c3c07cc0946f0eb37cf40e595f3deaac34c4bd7c6f1f4b1929880db08:0, height_hint=566232
2019-04-12 08:26:53.930 [INF] NTFN: New spend subscription: spend_id=7, outpoint=ca59f20c8a371678c4e2abaab4fbc0123e864cb2d39c0391c12afb120174b9b3:0, height_hint=566232
2019-04-12 08:26:53.930 [INF] NTFN: New spend subscription: spend_id=3, outpoint=a9c6512546094d3615587d1375459b516a43fa6f40aefb8d5e94211a1b998496:0, height_hint=566572
2019-04-12 08:26:54.084 [INF] CNCT: Close observer for ChannelPoint(b06ad896e955ab3fd92a870eafdac61ab300226492edcc800be2c27626778211:0) active
2019-04-12 08:26:54.192 [INF] CNCT: Close observer for ChannelPoint(b8e7a58656a00b85280a97a87e7f5455392dedefc7bb3089813d90b971ff9473:1) active
2019-04-12 08:26:54.303 [INF] CNCT: Close observer for ChannelPoint(4be8b1fd780739f20901d6bb0a6f3e17510026efbcf6b62f051463fa32dcb312:0) active
2019-04-12 08:26:54.412 [INF] CNCT: Close observer for ChannelPoint(f3665b0c3c07cc0946f0eb37cf40e595f3deaac34c4bd7c6f1f4b1929880db08:0) active
2019-04-12 08:26:54.521 [INF] CNCT: Close observer for ChannelPoint(ca59f20c8a371678c4e2abaab4fbc0123e864cb2d39c0391c12afb120174b9b3:0) active
2019-04-12 08:26:54.635 [INF] CNCT: Close observer for ChannelPoint(a9c6512546094d3615587d1375459b516a43fa6f40aefb8d5e94211a1b998496:0) active
2019-04-12 08:26:54.636 [INF] CNCT: ChannelArbitrator(ca59f20c8a371678c4e2abaab4fbc0123e864cb2d39c0391c12afb120174b9b3:0): starting state=StateCommitmentBroadcasted
2019-04-12 08:26:54.924 [INF] CNCT: ChannelArbitrator(ca59f20c8a371678c4e2abaab4fbc0123e864cb2d39c0391c12afb120174b9b3:0): noop trigger chainTrigger
2019-04-12 08:26:54.926 [INF] CNCT: ChannelArbitrator(cf91a4e77e81cdec69eb2fa94f8d15a4235cf60de06380a2aec4b743abd99091:1): starting state=StateWaitingFullResolution
2019-04-12 08:26:54.972 [INF] CNCT: ChannelArbitrator(cf91a4e77e81cdec69eb2fa94f8d15a4235cf60de06380a2aec4b743abd99091:1): still awaiting contract resolution
2019-04-12 08:26:54.972 [INF] CNCT: ChannelArbitrator(cf91a4e77e81cdec69eb2fa94f8d15a4235cf60de06380a2aec4b743abd99091:1): relaunching 1 contract resolvers
2019-04-12 08:26:54.972 [INF] NTFN: New confirmation subscription: txid=5a4d2eb780c0c4d3b07d9de57ed47411eed7bd9bfc65abc2fd59e747075a5a89, num_confs=1
2019-04-12 08:26:54.972 [INF] CNCT: ChannelArbitrator(b8e7a58656a00b85280a97a87e7f5455392dedefc7bb3089813d90b971ff9473:1): starting state=StateCommitmentBroadcasted
2019-04-12 08:26:54.972 [INF] NTFN: New confirmation subscription: conf_id=1, txid=5a4d2eb780c0c4d3b07d9de57ed47411eed7bd9bfc65abc2fd59e747075a5a89, height_hint=566760
2019-04-12 08:26:55.285 [INF] CNCT: ChannelArbitrator(b8e7a58656a00b85280a97a87e7f5455392dedefc7bb3089813d90b971ff9473:1): noop trigger chainTrigger
2019-04-12 08:26:55.287 [INF] CNCT: ChannelArbitrator(b06ad896e955ab3fd92a870eafdac61ab300226492edcc800be2c27626778211:0): starting state=StateCommitmentBroadcasted
2019-04-12 08:26:55.300 [INF] CNCT: ChannelArbitrator(b06ad896e955ab3fd92a870eafdac61ab300226492edcc800be2c27626778211:0): noop trigger chainTrigger
2019-04-12 08:26:55.300 [INF] CNCT: ChannelArbitrator(a9c6512546094d3615587d1375459b516a43fa6f40aefb8d5e94211a1b998496:0): starting state=StateCommitmentBroadcasted
2019-04-12 08:26:55.338 [INF] NTFN: Dispatching 1 confirmation notification for txid=5a4d2eb780c0c4d3b07d9de57ed47411eed7bd9bfc65abc2fd59e747075a5a89
2019-04-12 08:26:55.338 [INF] NTFN: New spend subscription: spend_id=9, outpoint=5a4d2eb780c0c4d3b07d9de57ed47411eed7bd9bfc65abc2fd59e747075a5a89:0, height_hint=571294
2019-04-12 08:26:55.338 [INF] CNCT: *contractcourt.commitSweepResolver(cf91a4e77e81cdec69eb2fa94f8d15a4235cf60de06380a2aec4b743abd99091:1): waiting for commit output to be swept
2019-04-12 08:26:55.576 [INF] CNCT: ChannelArbitrator(a9c6512546094d3615587d1375459b516a43fa6f40aefb8d5e94211a1b998496:0): noop trigger chainTrigger
2019-04-12 08:26:55.578 [INF] CNCT: ChannelArbitrator(f3665b0c3c07cc0946f0eb37cf40e595f3deaac34c4bd7c6f1f4b1929880db08:0): starting state=StateCommitmentBroadcasted
2019-04-12 08:26:55.856 [INF] CNCT: ChannelArbitrator(f3665b0c3c07cc0946f0eb37cf40e595f3deaac34c4bd7c6f1f4b1929880db08:0): noop trigger chainTrigger
2019-04-12 08:26:55.858 [INF] CNCT: ChannelArbitrator(4be8b1fd780739f20901d6bb0a6f3e17510026efbcf6b62f051463fa32dcb312:0): starting state=StateCommitmentBroadcasted
2019-04-12 08:26:56.102 [INF] CNCT: ChannelArbitrator(4be8b1fd780739f20901d6bb0a6f3e17510026efbcf6b62f051463fa32dcb312:0): noop trigger chainTrigger

```

**Comment by MykelSIlver** [comment]
> Closing for now since we've received no updates, and it appears to have been fixed by the recent PR I linked above (is part of 0.6).

lnd version 0.6.0-beta commit=v0.6-beta-rc4, which included fix #2766  did indeed solve the issue +1
